src/hooks/useApi.js

```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import { API_URL } from "../utils";
+
+export const useApi = () => {
+  const token = localStorage.getItem("token");
+  const callApi = async (method, endPoint, isProtected = false, data = {}) => {
+    if (isProtected) endPoint = `${API_URL}/${endPoint}`;
+    else {
+      endPoint = `${API_URL}/${endPoint}`;
+    }
+    switch (method) {
+      case "get": {
+        return await axios.get(
+          endPoint,
+          isProtected && {
+            headers: { authorization: token },
+          }
```

Comment:
suggestion: add this authorization part to axios interceptor

Reply:
not in every api call I need authorization